src/collapse/perf.rs

```diff
@@ -259,7 +259,8 @@ impl PerfState {
             }
 
             // --inline flag
-            if self.opt.show_inline && self.un_inline(pc, module) {
+            if self.opt.show_inline && can_un_inline(module) {
```

Comment:
This change doesn't seem right to me given what the commit message claims it does? Specifically, this will still fall back if we don't think we can un-inline? Is that intended behavior?

Reply:
Maybe the commit message just isn't clear?
This is what we're trying to match:
```
                 if ($show_inline == 1 && $mod !~ m/(perf-\d+.map|kernel\.|\[[^\]]+\])/) {
			unshift @stack, inline($pc, $mod);
			next;
		}
```
`can_inline(module)`  is meant to be roughly the same as `$mod !~ m/(perf-\d+.map|kernel\.|\[[^\]]+\])/`. So if it looks like something we can't un-inline we fall back to another resolution type. But if it looks like something we should be able to un-inline, but for whatever reason `un-inline` fails, we don't fall back.

Reply:
Ah, yes, you're entirely right!